tests/rptest/tests/schema_registry_test.py

```diff
@@ -3792,3 +3792,123 @@ def test_references(self):
 
         result = self.sr_client.get_schema(3)
         assert result == well_known_schema, f"Result: {result}"
+
+
+# dataset for SchemaRegistryCompatibilityModes: schemas is a list of 3 schemas compatible for `mode`, `antimode` is a suitable mode that will make the compat check for schemas fail
+CompatDataset = NamedTuple("CompatDataset", [("schemas", list[str]),
+                                             ("antimode", str | None)])
+
+
+class SchemaRegistryCompatibilityModes(SchemaRegistryEndpoints):
+    def __init__(self, test_context, **kwargs):
+        super().__init__(test_context, num_brokers=1, **kwargs)
+        self._sr_client = SchemaRegistryClient({'url': self._base_uri()})
+        self._topic = "test-topic"
+
+    def _register_schema(self,
+                         schema: str,
+                         type: str,
+                         mode: None | str,
+                         expect_invalid_schema=False):
+        """
+        utility to register a `schema` of `type` with compatibility set to `mode`.
+        if `expect_invalid_schema` is True, then a SchemaRegistryError is expected, for schema incompatibility
+        """
+        if mode is not None:
+            self._sr_client.set_compatibility(subject_name=self._topic,
+                                              level=mode)
+        try:
+            self._sr_client.register_schema(subject_name=self._topic,
+                                            schema=Schema(schema_str=schema,
+                                                          schema_type=type))
+            assert not expect_invalid_schema, "expected invalid schema"
+        except SchemaRegistryError as e:
+            assert expect_invalid_schema, f"{schema=} was expected valid, got {e=}"
+            assert e.error_code == 409, f"expected SchemaRegistryError 409/'Schema being registered is incompatible with an earlier schema [...]',  got {e=}"
+        except:
+            assert False, "unexpected exception"
```

Comment:
nit: this wouldn't show what the unexpected exception was, right? We could include it in the assertion message or just let it bubble up without a "catch-all".

Reply:
You are right, i can let it bubble up. I'll fix this, i have another commit to add anyway for avro

Reply:
done